Add bluechi is online tool

engelmi opened the pull request:
Relates to: #962 

So far, this PR adds the `bluechi-is-online` CLI tool, a man page and the RPM package to the spec file.

engelmi committed:
Implemented bluechi-is-online tool

Signed-off-by: Michael Engel <mengel@redhat.com>

engelmi committed:
Added new bluechi-is-online package and man page

Relates to: https://github.com/eclipse-bluechi/bluechi/issues/962

Signed-off-by: Michael Engel <mengel@redhat.com>

mkemel commented on doc/man/bluechi-is-online.1.md:
```diff
@@ -0,0 +1,76 @@
+
+## NAME
+
+bluechi-is-online - Command line tool to check and monitor the connection state of BlueChi's core components.
```
```suggestion
bluechi-is-online - Command line tool to check and monitor the connection state of Eclipse BlueChi™'s core components.
```

mkemel replied:
Do all the consequent occurrences should be `Eclipse BlueChi™` as well?

engelmi replied:
Its sufficient to have it once in the description (as we do in the other man pages). Added the basic description with the full name and TM symbol.

mkemel commented on doc/man/bluechi-is-online.1.md:
```diff
@@ -0,0 +1,76 @@
+**Options:**
+
+**--wait**
+Initial time to wait till the expected system status is `up`. In milliseconds.
```
```suggestion
Initial time in milliseconds to wait until the expected system status is `up`.
```
Same in the other occurrences of `--wait` description

engelmi replied:
Done.

mkemel replied:
@engelmi also `until` instead of `till`, "till" is more a spoken form

engelmi replied:
Done.

mkemel commented on doc/man/bluechi-is-online.1.md:
```diff
@@ -0,0 +1,76 @@
+
+## Commands
+
+### **bluechi-is-online** [*system*] [*options*]
```
```suggestion
### **bluechi-is-online** *system* [*options*]
```
The command method is not optional and not a parameter, so it shouldn't be in brackets. Same in other command examples

engelmi replied:
Done.

mkemel commented on src/is-online/help.c:
```diff
@@ -0,0 +1,38 @@
+void usage() {
+        usage_print_header();
+        usage_print_usage("bluechi-is-online [agent|node|system] [OPTIONS]");
+        printf("Available commands:\n");
```
Add the other commands as well?

engelmi replied:
Added :heavy_check_mark:

engelmi committed:
Added bluechi-is-online to readthedocs

Relates to: https://github.com/eclipse-bluechi/bluechi/issues/962

Added new section for tooling, moved the ansible page to it and
created bluechi-is-online page there, too.

Signed-off-by: Michael Engel <mengel@redhat.com>

mwperina commented on bluechi.spec.in:
```diff
@@ -244,6 +295,7 @@ will be used during integration tests when creating code coverage report.
 %endif
 
 
+
```
Do we need this 3rd empty line?

engelmi replied:
No, I added it just to have a bit of "separation" between defining the packages and the actual build steps.

mkemel commented on src/is-online/is-online.c:
```diff
@@ -0,0 +1,469 @@
+        return r >= 0;
+}
+
+static int get_options(
```
Maybe it would be better to extend the `command_get_option` interface in command.h and add `command_get_option_long` function, that would do the parsing, print errors? Then we wouldn't need this function, and the whole long option handling would be come easier.

engelmi replied:
This is the only location for long parsing of CLI option values at the moment, but long values is a common use case - added the `command_get_option_long`, but without the error logging. The caller can provide a better log message incl. the arg name (instead of the key). 
When implementing this, I was able to reduce the amount of code quite a bit by moving the common code of all three is-online-functions to a single, parameterized one.

engelmi replied:
Added missing unit tests for `command_get_option_long`.

mkemel replied:
That was my intention
Looks great now!

ArtiomDivak reviewed (approved):
LGTM

mkemel reviewed (approved):
LGTM
